### PR TITLE
Add Eicfunctions and GLN to Token

### PR DESCRIPTION
**Pull request by FirestarJes**
This will add the eicfunctions to the token instead of the currently incorrect data.
It will also add the GLN number to the token

### CODE REVIEW

File: source/Energinet.DataHub.MarketParticipant.Domain/Services/Rules/OverlappingEicFunctionsRuleService.cs [hunk 1]
```diff
@@ -20,16 +20,13 @@
 
 namespace Energinet.DataHub.MarketParticipant.Domain.Services.Rules
 {
-    public sealed class OverlappingBusinessRolesRuleService : IOverlappingBusinessRolesRuleService
+    public sealed class OverlappingEicFunctionsRuleService : IOverlappingEicFunctionsRuleService
     {
-        private readonly IBusinessRoleCodeDomainService _businessRoleCodeDomainService;
-
-        public OverlappingBusinessRolesRuleService(IBusinessRoleCodeDomainService businessRoleCodeDomainService)
+        public OverlappingEicFunctionsRuleService()
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.MarketParticipant.Domain/Services/Rules/OverlappingEicFunctionsRuleService.cs:25]: Can be deleted now right?

File: source/Energinet.DataHub.MarketParticipant.Infrastructure/Services/ActiveDirectoryB2CRolesProvider.cs [hunk 2]
```diff
@@ -58,38 +59,54 @@ public async Task<ActiveDirectoryB2CRoles> GetB2CRolesAsync()
                 switch (appRole.Value)
                 {
                     case "balanceresponsibleparty":
-                        _activeDirectoryB2CRoles.DdkId = appRole.Id!.Value;
+                        _activeDirectoryB2CRoles.EicRolesMapped.Add(EicFunction.BalanceResponsibleParty, appRole.Id!.Value);
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.MarketParticipant.Infrastructure/Services/ActiveDirectoryB2CRolesProvider.cs:62]: You could do something like this, and just handle the special cases, if any (none as far as I can see)
```c#
var lookup = ((EicFunction[])typeof(EicFunction).GetEnumValues()).ToDictionary(x => x.ToString().ToLowerInvariant());

foreach (var appRole in application.AppRoles)
{
    if (lookup.TryGetValue(appRole.Value, out var val))
    {
        _activeDirectoryB2CRoles.EicRolesMapped.Add(val, appRole.Id!.Value);
    }

    throw new ArgumentOutOfRangeException();
}

```

Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Infrastructure/Services/ActiveDirectoryB2CRolesProvider.cs:62]: Nice one :)

Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Infrastructure/Services/ActiveDirectoryB2CRolesProvider.cs:62]: Fixed

File: source/Energinet.DataHub.MarketParticipant.Domain/Services/Rules/OverlappingEicFunctionsRuleService.cs (hunk 1 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Domain/Services/Rules/OverlappingEicFunctionsRuleService.cs:25]: Fixed

**Review by michaelmajgaard — APPROVED** [review]

**Pull request merged**